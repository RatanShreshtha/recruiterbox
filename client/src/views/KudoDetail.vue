<template>
  <b-container class="p-4" fluid>
    <b-jumbotron>
      <template v-slot:header>
        {{ kudo.title }}
        <hr class="my-4" />
      </template>

      <template v-slot:lead>
        <div class="row">
          <div class="col-4 text-left">
            <strong>From: </strong>{{ kudo.by_user.first_name }}
            {{ kudo.by_user.last_name }} ({{ kudo.by_user.username }})
          </div>
          <div class="col-4 text-center">
            <strong>To: </strong>{{ kudo.to_user.first_name }}
            {{ kudo.to_user.last_name }} ({{ kudo.to_user.username }})
          </div>
          <div class="col-4 text-right">
            {{ kudo.awarded | moment("calendar") }}
          </div>
        </div>
        <br />
        {{ kudo.description }}
      </template>
    </b-jumbotron>
  </b-container>
</template>

<script>
import { mapState } from "vuex";

import { apiService } from "@/service";
export default {
  name: "KudoDetail",
  computed: {
    ...mapState(["token"])
  },
  props: {
    kudoId: {
      type: Number,
      required: true
    }
  },
  data() {
    return {
      kudo: null
    };
  },
  async mounted() {
    this.getKudo();
  },
  methods: {
    async getKudo() {
      try {
        const resp = await apiService.get(`/api/kudos/${this.kudoId}`, {
          headers: {
            Authorization: `Token ${this.token}`
          }
        });
        this.kudo = resp.data;
      } catch (error) {
        console.log("error", error);
      }
    }
  }
};
</script>

<style scoped></style>
