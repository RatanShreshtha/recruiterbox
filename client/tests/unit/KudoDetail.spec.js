import { shallowMount, createLocalVue } from "@vue/test-utils";

import { createRenderer } from "vue-server-renderer";

import Vuex from "vuex";
import VueRouter from "vue-router";
import { BootstrapVue, IconsPlugin } from "bootstrap-vue";
import createPersistedState from "vuex-persistedstate";

import KudoDetail from "@/views/KudoDetail.vue";

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueRouter);

localVue.use(BootstrapVue);
localVue.use(IconsPlugin);

describe("KudoDetail.vue", () => {
  let propsData;
  let state;
  let store;
  let routes;
  let router;

  beforeEach(() => {
    propsData = {
      kudoId: 1
    };
    state = {
      id: null,
      kudos: 0,
      username: null,
      first_name: null,
      last_name: null,
      email: null,
      token: null
    };

    store = new Vuex.Store({
      plugins: [createPersistedState()],
      state
    });

    routes = [
      {
        path: "/",
        name: "Home"
      }
    ];

    router = new VueRouter({
      routes
    });
  });
  it("matches snapshot", () => {
    const renderer = createRenderer();

    const wrapper = shallowMount(KudoDetail, {
      localVue,
      store,
      propsData,
      router
    });

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
