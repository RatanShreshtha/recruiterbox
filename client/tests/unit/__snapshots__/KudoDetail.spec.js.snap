// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KudoDetail.vue matches snapshot 1`] = `
<b-container-stub tag="div" fluid="true" class="p-4">
  <b-jumbotron-stub headerTag="h1" headerLevel="3" leadTag="p" tag="div"></b-jumbotron-stub>
</b-container-stub>
`;
